agent/consul/state/catalog.go

```diff
@@ -1106,6 +1108,9 @@ func (s *Store) AssignManualServiceVIPs(idx uint64, psn structs.PeeredServiceNam
 	for _, ip := range ips {
 		assignedIPs[ip] = struct{}{}
 	}
+
+	txnNeedsCommit := false
```

Comment:
I don't think this is practically an issue, but I did notice that the logic was:
```
begin txn
maybe write
maybe early return
write
commit
```
and with this change i fixed it to
```
begin txn
maybe write
maybe write
maybe commit
```